servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DnsServiceDiscovererObserver.java

```diff
@@ -30,9 +30,23 @@ public interface DnsServiceDiscovererObserver {
      * @param name the name of DNS record to be discovered
      * @return {@link DnsDiscoveryObserver} that provides visibility into individual DNS resolutions behind the
      * associated discovery
+     * @deprecated use {@link #onNewDiscovery(String, String)} instead.
      */
+    @Deprecated // FIXME 0.43: remove deprecated method
```

Comment:
Because `DnsServiceDiscovererObserver` is a `@FunctionalInterface` right now, we won't be able to simply remove this method in the next release. Consider describing the recommended approach for users, similar to this example: https://github.com/apple/servicetalk/pull/2525/files#diff-5d32e457fd45a95d48cdea9552a11778b7063fe42b5ff9ef6ea5c792a28c42e9R74-R77